extensions/amazon-aws/src/components/searchbar/aws-profile-dropdown.tsx

```diff
@@ -1,16 +1,32 @@
-import { Icon, List } from "@raycast/api";
+import { Icon, List, getPreferenceValues } from "@raycast/api";
 import { useEffect } from "react";
 import { loadSharedConfigFiles } from "@aws-sdk/shared-ini-file-loader";
 import { useCachedPromise, useCachedState, useExec } from "@raycast/utils";
 
+interface Preferences {
+  useAWSVault: boolean;
+}
 interface Props {
   onProfileSelected?: VoidFunction;
 }
 
 export default function AWSProfileDropdown({ onProfileSelected }: Props) {
+  const preferences = getPreferenceValues<Preferences>();
   const [selectedProfile, setSelectedProfile] = useCachedState<string>("aws_profile");
   const profileOptions = useProfileOptions();
 
+  let isUsingAwsVault = false;
+  let vaultSessions: string[] | undefined = undefined;
+
+  if (preferences.useAWSVault) {
+    vaultSessions = useVaultSessions();
+    isUsingAwsVault = !!vaultSessions;
+    useAwsVault({
+      profile: vaultSessions?.includes(selectedProfile || "") ? selectedProfile : undefined,
+      onUpdate: () => onProfileSelected?.(),
+    });
+  }
```

Comment:
Calling hooks conditionally is [not allowed](https://legacy.reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level) and might lead to unexpected behavior.

Instead, I suggest using the `execute` config field of the `useExec` call in the `useAwsVault` hook. If `execute` is `false` in that hook, this should be an effective way to disable aws-vault.

Reply:
fixed, thanks